src/gt4py/next/iterator/ir_utils/ir_makers.py

```diff
@@ -389,3 +396,34 @@ def _impl(*its: itir.Expr) -> itir.FunCall:
 def map_(op):
     """Create a `map_` call."""
     return call(call("map_")(op))
+
+
+def cartesian_domain(ranges: Dict[str, Tuple[Any, Any]], out_field: str,
+                     size_field: list[itir.Literal] = None) -> SymbolicDomain:
+    axis_order = ['IDim', 'JDim', 'KDim']
+
+    if size_field is not None:
+        assert len(ranges) == len(size_field)
+    else:
+        size_field = [None] * len(ranges)
+
+    symbolic_ranges = {}
+
+    for index, (axis, (start, stop)) in enumerate(ranges.items()):
+        if axis not in axis_order:
+            raise ValueError("The ranges need to contain either IDim, JDim or KDim.")
+
+        axis_index = axis_order.index(axis)
+        start_expr = ConstantFolding.apply(start)
```

Comment:
You only need the constant folding for readability in your tests so this should happen there. The makers should not contain any calls to passes or "optimizations", but just be simple tools that ease creation of nodes.